modules/storage/filesystem/device.go

```diff
@@ -0,0 +1,186 @@
+package filesystem
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/rs/zerolog/log"
+)
+
+type DeviceManager interface {
+	Device(ctx context.Context, device string) (Device, error)
+	Devices(ctx context.Context) ([]Device, error)
+	WithLabel(ctx context.Context, label string) ([]Device, error)
+	PoolType(ctx context.Context, pool Pool) (DeviceType, error)
+}
+
+// FSType type of filesystem on device
+type FSType string
+
+const (
+	// BtrfsFSType btrfs filesystem type
+	BtrfsFSType FSType = "btrfs"
+)
+
+// DeviceType is the actual type of hardware that the storage device runs on,
+// i.e. SSD or HDD
+type DeviceType string
+
+// Known device types
+const (
+	SSDDevice = "SSD"
+	HDDDevice = "HDD"
+)
+
+// Device represents a physical device
+type Device struct {
+	Type       string   `json:"type"`
+	Path       string   `json:"name"`
+	Label      string   `json:"label"`
+	Filesystem FSType   `json:"fstype"`
+	Children   []Device `json:"children"`
+	DiskType   DeviceType
```

Comment:
Please add the json tags for consistency, if u want it getting filled up from the lsblk output put `json:"-"`.